java/arcs/sdk/jvm/CollectionImpl.kt

```diff
@@ -11,35 +11,37 @@
 
 package arcs.sdk
 
+import arcs.core.common.Referencable
+import arcs.core.crdt.CrdtSet
+
 /** [ReadWriteCollection] implementation for the JVM. */
-@Suppress("UNUSED_PARAMETER")
-// TODO: Connect to storage.
-class CollectionImpl<T : Entity>(
-    val particle: Particle,
+class CollectionImpl<T>(
     override val name: String,
-    entitySpec: EntitySpec<T>
-) : ReadWriteCollection<T> {
-    private val entities = mutableListOf<T>()
-
+    particle: Particle,
+    storageProxy: StorageProxy<CrdtSet.Data<T>, CrdtSet.IOperation<T>, Set<T>>
+) : HandleImpl<CrdtSet.Data<T>, CrdtSet.IOperation<T>, Set<T>>(name, particle, storageProxy), ReadWriteCollection<T> where T : Entity, T : Referencable {
     override val size: Int
-        get() = entities.size
+        get() = storageProxy.getParticleView().value.size
 
-    override fun isEmpty(): Boolean = entities.isEmpty()
+    override fun isEmpty(): Boolean = storageProxy.getParticleView().value.isEmpty()
 
-    override fun iterator(): Iterator<T> = entities.iterator()
+    override fun iterator(): Iterator<T> = storageProxy.getParticleView().value.iterator()
 
     override fun store(entity: T) {
-        entities.add(entity)
+        clock.increment()
+        storageProxy.applyOp(CrdtSet.Operation.Add(clock, name, entity))
         particle.onHandleUpdate(this)
     }
 
     override fun clear() {
-        entities.clear()
+        storageProxy.getParticleView().value.forEach {
+            storageProxy.applyOp(CrdtSet.Operation.Remove(clock, name, it))
```

Comment:
I'd be curious to hear about batching here. Are the ops going to be applied one at a time, or will the storage proxy / store do its own batching?

Maybe just add a TODO to investigate?

Reply:
Good question! This appears (?) to match the behavior of the storageNG handles on the TS side, which are what guide me.